src/python/pants/backend/python/macros/python_requirements.py

```diff
@@ -31,6 +37,40 @@
 from pants.util.strutil import softwrap
 
 
+def parse_pyproject_toml(
```

Comment:
Nit, we typically prefer to have the target definitions at the top of the file, and rule related code beneath it.